CMakeLists.txt

```diff
@@ -97,6 +91,13 @@ IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     ELSE()
         MESSAGE(WARNING "\nCannot determine the version of the compiler selected to build GNU Radio (GCC : ${CMAKE_CXX_COMPILER}). This build may or not work. We highly recommend using GCC version ${GCC_MIN_VERSION} or more recent.")
     ENDIF()
+    # Release flags: -O3 enables tree vectorization
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O2")
+    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
```

Comment:
``-O2 -g -DNDEBUG`` is the default for ``RELWITHDEBINFO`` in CMake if I remember correctly. ``-DNDEBUG`` is definitely a default which I find very, very questionable – I have been bitten by this several times…

Reply:
Yes `-O2 -g -DNDEBUG` is the default even on distributions like Fedora. The `-DNDEBUG` is mainly used to remove all `assert` checks from the result binaries and in my opinion it is desired flag for release builds.

Reply:
Are you two saying you would prefer to set this to `-O2`? I would be OK with that, although I'm not sure who uses this build setting. Do distro maintainers use this to generate debug info? In that case, we'd be shipping O2-level binaries to all Ubuntu/Debian users.

Reply:
I'd prefer the `RELWITHDEBINFO` settings here and below to be `-O2` ... should make debugging reasonable since those optimizations mostly follow the code ... while `-O3` does a lot of stuff that make the resulting binary objects not follow the code so closely.

Reply:
ping on this change request, here and for Clang too

Reply:
[x]